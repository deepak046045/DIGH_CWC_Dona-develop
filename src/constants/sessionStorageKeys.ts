export const sessionStorageKeys = {
  IsUpcomingAppointmentAlertDismissed: 'isUpcomingAppointmentBannerDismissed',
};
