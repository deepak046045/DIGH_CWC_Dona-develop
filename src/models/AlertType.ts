export type AlertType = {
  isOpen: boolean;
  message: string;
  level: string;
};
