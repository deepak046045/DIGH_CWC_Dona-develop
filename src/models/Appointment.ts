import { AppointmentType } from './AppointmentType';
import { CarePartner } from './CarePartner';

export type Appointment = {
  id: string;
  date: Date;
  address: string;
  type: AppointmentType;
  carePartners: CarePartner[];
};
