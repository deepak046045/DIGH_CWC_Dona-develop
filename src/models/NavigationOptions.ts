export type NavigationOption = {
  title: string;
  path: string;
};

export const NavigationOptions: Record<string, NavigationOption> = {
  TreatmentMap: {
    title: 'Treatment map',
    path: '/treatmentMap',
  },
  WhatToExpect: {
    title: 'What to expect',
    path: '/whatToExpect',
  },
  MyTreatmentPlan: {
    title: 'My treatment plan',
    path: '/myTreatmentPlan',
  },
  MyAppointments: {
    title: 'My appointments',
    path: '/myAppointments',
  },
  AddAppointment: {
    title: 'Add appointment',
    path: '/addAppointment',
  },
  MyCarePartners: {
    title: 'My care partners',
    path: '/myCarePartners',
  },
  MyProfile: {
    title: 'My profile',
    path: '/myProfile',
  },
};

export const gatedNavigationOptions: NavigationOption[] = [
  NavigationOptions.MyTreatmentPlan,
  NavigationOptions.MyAppointments,
  NavigationOptions.MyCarePartners,
  NavigationOptions.WhatToExpect,
];

export const ungatedNavigationOptions: NavigationOption[] = [
  NavigationOptions.TreatmentMap,
  NavigationOptions.WhatToExpect,
];
