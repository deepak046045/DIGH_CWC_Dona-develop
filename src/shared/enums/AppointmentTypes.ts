export enum AppointmentTypes {
  INFUSION = 'infusion',
  SAFETY_MRI = 'safetyMRI',
  NEUROLOGIST = 'neurologist',
  GENERAL_DOCTOR = 'generalDoctor',
}
