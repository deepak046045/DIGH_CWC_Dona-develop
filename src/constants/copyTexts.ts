import mockUserData from '@/mocks/mockUserData.json';
import { AppointmentTypes } from '@/shared/enums/AppointmentTypes';

export const copyTexts = {
  WelcomeSection: {
    welcomeHeaderTextUngated: `Welcome to Kisunla support.`,
    welcomeHeaderTextGated: `Hi ${mockUserData.firstName}, welcome to Kisunla support`,
    welcomeSubHeaderTextUngated: `Support resources for your treatment journey. Register to set up a
      calendar of appointments, set up reminders, and assign care
      partners.`,
    welcomeSubHeaderTextGated: `Support resources for your treatment journey. Set up your calendar of appointments, reminders, and care partners.`,
  },
  MyAppointments: {
    header: 'My appointments',
    subHeader:
      'Appointment reminders will be sent to you and your care partners 5 days and 1 day before your scheduled appointment.',
    noAppointments: {
      firstPiece: 'No appointments saved.',
      link: 'Add New Appointments',
      secondPiece: 'to get started.',
    },
    noUpcomingAppointments: {
      firstPiece: 'You have no upcoming appointments.',
      addLink: 'Add new appointments',
      or: 'or',
      viewLink: 'view all Appointments',
    },
    tableColumnHeaders: {
      date: 'Date',
      time: 'Time',
      address: 'Address',
      appointmentType: 'Type of appointment',
      carePartners: 'Care partner(s)',
      actions: 'Actions',
    },
    buttons: {
      newAppointment: 'New appointment',
      edit: 'Edit',
      delete: 'Delete',
    },
    deleteModal: {
      header: 'Are you sure you want to delete this appointment?',
      subHeader: 'This cannot be undone.',
      primaryButtonLabel: 'Yes, delete',
      secondaryButtonLabel: 'Cancel',
    },
    deleteAlert: {
      message: 'The appointment has been successfully deleted.',
    },
    appointmentTypes: {
      [AppointmentTypes.INFUSION]: 'IV infusion Dose',
      [AppointmentTypes.GENERAL_DOCTOR]: 'General Practicioner Check-in',
      [AppointmentTypes.SAFETY_MRI]: 'Safety MRI',
      [AppointmentTypes.NEUROLOGIST]: 'Neurologist Check-in',
    },
  },
  TreatmentIntro: {
    titleGated: 'My treatment plan',
    titleUngated: 'Recommended treatment',
    subtitleUngated:
      'Setting you up for success in your treatment. Setting you up for success in your treatment.',
    textUngated:
      'This treatment map serves as a suggested guide. Always follow your doctor’s directions. Insurance approval may be needed or the start of your infusion therapy may be delayed.',
    addApointmentBtnGated: 'Add Appointment',
  },
  TreatmentDownload: {
    reminderUngated:
      'Always follow your doctor’s directions. Insurance approval may be needed or the start of your infusion therapy may be delayed.',
    downloadUngated:
      'For a printable version of the treatment map download the following.',
    downloadGated: 'Print or save your Treatment Plan as a PDF',
    downloadButtonUngated: 'Treatment Map PDF',
    downloadButtonGated: 'Print My Plan',
  },
  Header: {
    signIn: 'Sign in',
    signUp: 'Sign up',
    signOut: 'Sign out',
    viewAppointments: 'View your appointments',
    nextAppointmentToday: 'Your next appointment is today.',
    nextAppointmentIsIn: 'Your next appointment is in',
    day: 'day',
    days: 'days',
  },
  Footer: {
    primaryLinks: {
      prescribingInformation: 'Prescribing Information',
      medicationGuide: 'Medication Guide',
    },
    globalLinks: {
      privacyPolicy: 'Privacy policy',
      termsOfUse: 'Terms of use',
      accessibilityStatement: 'Accessibility Statement',
    },
    saveToDeviceHomeButton: 'Save to device home',
  },
  AddAppointment: {
    title: 'Add Appointment',
    subtitle:
      'Appointment reminders will be sent to you and your care partners 5 days and 1 day before your scheduled appointment.',
    selectOptionLabels: {
      [AppointmentTypes.INFUSION]: 'Infusion',
      [AppointmentTypes.NEUROLOGIST]: 'Neurologist',
      [AppointmentTypes.SAFETY_MRI]: 'Safety MRI',
      [AppointmentTypes.GENERAL_DOCTOR]: 'General Doctor',
    },
    labels: {
      appointmentType: 'TYPE OF APPOINTMENT',
      infusionNumber: 'INFUSION # (OPTIONAL)',
      address: 'ADDRESS',
      date: 'DATE',
      time: 'TIME',
      carePartners: 'CARE PARTNERS (OPTIONAL)',
    },
    hints: {
      date: 'Format: MM/DD/YYYY',
      time: 'Format: HH:MM',
    },
    formErrors: {
      appointmentType: 'Select an appointment type',
      fieldRequired: 'This field is required',
      addressExceedLimit: 'Address exceeds 150 characters', // TODO: define this copy
      invalidDate: 'Date is invalid',
      invalidTime: 'Time is invalid',
    },
    confirmationModal: {
      title: 'Your appointment has been saved.',
      primaryButtonLabel: 'View my appointments',
      secondaryButtonLabel: 'Add new appointment',
    },
    addCarePartnerBtn: 'Add new care partner',
    saveBtn: 'Save',
    cancelBtn: 'Cancel',
  },
  AddCarePartnerModal: {
    title: 'Add care partner',
    name: 'NAME',
    email: 'EMAIL ADDRESS',
    cancelButton: 'Cancel',
    saveButton: 'Save',
    notSavedWarning: 'You have not saved your care partner.',
    allDetailsLost: 'All details will be lost.',
    closeAnywayButton: 'Close anyway',
    formValidation: {
      requiredField: 'This field is required',
      invalidEmail: 'Invalid email address',
      nameLength: 'Name must not be longer than 50 characters',
      emailLength: 'Email address must not be longer than 80 characters',
    },
  },
};
