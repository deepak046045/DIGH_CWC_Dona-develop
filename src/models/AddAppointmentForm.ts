import { AddAppointmentFormFields } from '@/shared/enums/AddAppointmentFormFields';
import { AppointmentTypes } from '@/shared/enums/AppointmentTypes';

export type AddAppointmentForm = {
  [AddAppointmentFormFields.APPOINTMENT_TYPE]: AppointmentTypes | '';
  [AddAppointmentFormFields.INFUSION_NUMBER]: string;
  [AddAppointmentFormFields.ADDRESS]: string;
  [AddAppointmentFormFields.SCHEDULE_DATE]: string;
  [AddAppointmentFormFields.SCHEDULE_TIME]: number;
  [AddAppointmentFormFields.CARE_PARTNERS]: string[];
};
