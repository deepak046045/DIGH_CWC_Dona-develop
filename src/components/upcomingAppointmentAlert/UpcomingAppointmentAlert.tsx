'use client';

import React, { useContext, useEffect, useState } from 'react';
import { LdsButton, LdsIcon, LdsLink } from '@elilillyco/ux-lds-react';
import { useRouter } from 'next/navigation';
import { UserContext } from '@/context/UserContextProvider';
import { AlertType } from '@/models/AlertType';
import { copyTexts } from '@/constants/copyTexts';
import { sessionStorageKeys } from '@/constants/sessionStorageKeys';
import alertStyles from '@/scss/themes/alert.module.scss';
import { NavigationOptions } from '@/models/NavigationOptions';
import {
  removeSessionStorageItem,
  setSessionStorageItem,
} from '@/utils/sessionStorage';
import { getUpcomingAppointmentAlertMessage } from '@/utils/upcomingAppointmentAlert';
import styles from './UpcomingAppointmentAlert.module.scss';

export function UpcomingAppointmentAlert() {
  const { isLoggedIn, appointments, setActiveView } = useContext(UserContext);
  const [alert, setAlert] = useState<AlertType>({
    isOpen: false,
    message: '',
    level: 'info',
  });

  const router = useRouter();

  function setUpcomingAppointmentAlert(): void {
    const message = getUpcomingAppointmentAlertMessage(appointments);

    if (message) {
      setAlert({
        isOpen: true,
        message,
        level: 'info',
      });
    }
  }

  const handleAlertClose = () => {
    setAlert({ ...alert, isOpen: false });
    setSessionStorageItem(
      sessionStorageKeys.IsUpcomingAppointmentAlertDismissed,
      true
    );
  };

  useEffect(() => {
    if (isLoggedIn) {
      setUpcomingAppointmentAlert();
    } else {
      removeSessionStorageItem(
        sessionStorageKeys.IsUpcomingAppointmentAlertDismissed
      );
      setAlert({ ...alert, isOpen: false });
    }
  }, [isLoggedIn, appointments]);

  return (
    <div id="test" className={`${alertStyles.primary}`}>
      {alert.isOpen && (
        <div
          id="upcoming-appointment-alert"
          role="alert"
          aria-live="polite"
          className="lds-alert info"
        >
          <div className="lds-alert-message">
            <LdsButton
              data-testid="dismiss-upcoming-appointment-banner"
              clearDefaultClasses
              classes="lds-button-base lds-alert-button"
              onClick={handleAlertClose}
            >
              <LdsIcon
                className={styles.closeIconWhite}
                label="close"
                name="X"
                inline
              />
            </LdsButton>
            <div className={styles.alert}>
              <div className={styles.infoIcon}>
                <LdsIcon
                  classes="lds-alert-icon info"
                  label="info"
                  name="InfoFill"
                />
              </div>
              <span className={styles.alertContent}>
                {alert.message}&nbsp;
                <LdsLink
                  data-testid="view-appointments-banner-link"
                  classes={styles.upcomingAppointmentAlertLink}
                  onClick={() => {
                    router.push(NavigationOptions.MyAppointments.path);
                    setActiveView(NavigationOptions.MyAppointments);
                  }}
                >
                  {copyTexts.Header.viewAppointments}
                </LdsLink>
              </span>
            </div>
          </div>
        </div>
      )}
    </div>
  );
}
