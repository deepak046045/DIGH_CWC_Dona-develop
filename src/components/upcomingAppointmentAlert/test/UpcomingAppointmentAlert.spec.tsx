import { fireEvent, render, screen } from '@testing-library/react';
import { NavigationOptions } from '@/models/NavigationOptions';
import {
  UserContextProvider,
  UserContext,
} from '@/context/UserContextProvider';
import React from 'react';
import { AppointmentTypes } from '@/shared/enums/AppointmentTypes';
import '@testing-library/jest-dom';
import { sessionStorageKeys } from '../../../constants/sessionStorageKeys';
import {
  getSessionStorageItem,
  setSessionStorageItem,
} from '../../../utils/sessionStorage';
import { UpcomingAppointmentAlert } from '../UpcomingAppointmentAlert';

jest.mock('next/navigation', () => ({
  useRouter: jest.fn(() => ({ push: jest.fn() })),
}));
jest.mock('crypto', () => ({ randomUUID: () => '123456789' }));

describe('UpcomingAppointmentAlert Component', () => {
  it('should show upcoming appointment alert banner for 14 days from now', () => {
    const mockValue = {
      isLoggedIn: true,
      appointments: [
        {
          id: '1',
          date: new Date(Date.now() + 14 * 24 * 60 * 60 * 1000),
          address: '11700 N. Meridian Street, Carmel, IN 46032',
          type: {
            name: AppointmentTypes.INFUSION,
            count: 2,
          },
          carePartners: ['Henry', 'Charlotte'],
        },
      ],
      setIsLoggedIn: jest.fn(),
      setActiveView: jest.fn(),
      setUpcomingAppointmentAlert: jest.fn(),
    };

    const { container } = render(
      <UserContext.Provider value={mockValue}>
        <UpcomingAppointmentAlert />
      </UserContext.Provider>
    );

    expect(
      container.querySelector('#upcoming-appointment-alert')
    ).toBeInTheDocument();
    expect(container).toMatchSnapshot();
  });

  it('should show upcoming appointment alert banner for 1 day from now', () => {
    const mockValue = {
      isLoggedIn: true,
      appointments: [
        {
          id: '1',
          date: new Date(Date.now() + 1 * 24 * 60 * 60 * 1000),
          address: '11700 N. Meridian Street, Carmel, IN 46032',
          type: {
            name: AppointmentTypes.INFUSION,
            count: 2,
          },
          carePartners: ['Henry', 'Charlotte'],
        },
      ],
      setIsLoggedIn: jest.fn(),
      setActiveView: jest.fn(),
      setUpcomingAppointmentAlert: jest.fn(),
    };

    const { container } = render(
      <UserContext.Provider value={mockValue}>
        <UpcomingAppointmentAlert />
      </UserContext.Provider>
    );

    expect(container).toMatchSnapshot();
  });

  it('should not show upcoming appointment alert banner if session store confirms that its already been dismissed', () => {
    setSessionStorageItem(
      sessionStorageKeys.IsUpcomingAppointmentAlertDismissed,
      true
    );
    const mockValue = {
      isLoggedIn: true,
    };
    const { container } = render(
      <UserContextProvider value={mockValue}>
        <UpcomingAppointmentAlert />
      </UserContextProvider>
    );

    expect(container.querySelector('#upcoming-appointment-alert')).toBeNull();
    window.sessionStorage.clear();
  });

  it('should not show upcoming appointment alert banner if no upcoming appointments', () => {
    const mockValue = {
      appointments: [],
      isLoggedIn: true,
      setIsLoggedIn: jest.fn(),
      setActiveView: jest.fn(),
      setUpcomingAppointmentAlert: jest.fn(),
    };

    const { container } = render(
      <UserContext.Provider value={mockValue}>
        <UpcomingAppointmentAlert />
      </UserContext.Provider>
    );

    expect(container.querySelector('#upcoming-appointment-alert')).toBeNull();
  });

  it('should show specific banner if next appointment is today', () => {
    const mockValue = {
      isLoggedIn: true,
      appointments: [
        {
          id: '1',
          date: new Date(Date.now() + 1 * 60 * 60 * 1000),
        },
      ],
      setIsLoggedIn: jest.fn(),
      setActiveView: jest.fn(),
      setUpcomingAppointmentAlert: jest.fn(),
    };

    const { container } = render(
      <UserContext.Provider value={mockValue}>
        <UpcomingAppointmentAlert />
      </UserContext.Provider>
    );

    expect(container).toMatchSnapshot();
  });

  it('should dismiss banner and set sessionStorage when close clicked', () => {
    const mockValue = {
      isLoggedIn: true,
      appointments: [
        {
          id: '1',
          date: new Date(Date.now() + 14 * 24 * 60 * 60 * 1000),
        },
      ],
      setIsLoggedIn: jest.fn(),
      setActiveView: jest.fn(),
      setUpcomingAppointmentAlert: jest.fn(),
    };

    const { container } = render(
      <UserContext.Provider value={mockValue}>
        <UpcomingAppointmentAlert />
      </UserContext.Provider>
    );

    const closeButton = screen.getByTestId(
      'dismiss-upcoming-appointment-banner'
    );
    fireEvent.click(closeButton);

    expect(container.querySelector('#upcoming-appointment-alert')).toBeNull();
    expect(
      getSessionStorageItem(
        sessionStorageKeys.IsUpcomingAppointmentAlertDismissed
      )
    ).toEqual(true);
  });

  it('should navigate to myAppointments when view appointments banner link clicked', () => {
    window.sessionStorage.clear();
    const mockSetActiveView = jest.fn();
    const mockValue = {
      isLoggedIn: true,
      appointments: [
        {
          id: '1',
          date: new Date(Date.now() + 14 * 24 * 60 * 60 * 1000),
          address: '11700 N. Meridian Street, Carmel, IN 46032',
          type: {
            name: AppointmentTypes.INFUSION,
            count: 2,
          },
          carePartners: ['Henry', 'Charlotte'],
        },
      ],
      setIsLoggedIn: jest.fn(),
      setActiveView: mockSetActiveView,
      setUpcomingAppointmentAlert: jest.fn(),
    };

    render(
      <UserContext.Provider value={mockValue}>
        <UpcomingAppointmentAlert />
      </UserContext.Provider>
    );

    const viewAppointmentsLink = screen.getByTestId(
      'view-appointments-banner-link'
    );
    fireEvent.click(viewAppointmentsLink);

    expect(mockSetActiveView).toHaveBeenCalledWith(
      NavigationOptions.MyAppointments
    );
  });
});
