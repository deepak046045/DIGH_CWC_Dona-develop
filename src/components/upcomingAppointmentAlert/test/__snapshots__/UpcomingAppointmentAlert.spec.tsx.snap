// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UpcomingAppointmentAlert Component should show specific banner if next appointment is today 1`] = `
<div>
  <div
    class="primary"
    id="test"
  >
    <div
      aria-live="polite"
      class="lds-alert info"
      id="upcoming-appointment-alert"
      role="alert"
    >
      <div
        class="lds-alert-message"
      >
        <button
          class="lds-button-base lds-alert-button"
          data-testid="dismiss-upcoming-appointment-banner"
          type="button"
        >
          <svg
            aria-labelledby="lds-icon-X-title-123456789 lds-icon-X-description-123456789"
            class="lds-icon icon-X inline closeIconWhite"
            focusable="false"
            role="img"
          >
            <title
              id="lds-icon-X-title-123456789"
            >
              close
            </title>
            <desc
              id="lds-icon-X-description-123456789"
            />
            <svg
              viewBox="0 0 256 256"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M205.66,194.34a8,8,0,0,1-11.32,11.32L128,139.31,61.66,205.66a8,8,0,0,1-11.32-11.32L116.69,128,50.34,61.66A8,8,0,0,1,61.66,50.34L128,116.69l66.34-66.35a8,8,0,0,1,11.32,11.32L139.31,128Z"
              />
            </svg>
          </svg>
        </button>
        <div
          class="alert"
        >
          <div
            class="infoIcon"
          >
            <svg
              aria-labelledby="lds-icon-InfoFill-title-123456789 lds-icon-InfoFill-description-123456789"
              class="lds-icon icon-InfoFill"
              classes="lds-alert-icon info"
              focusable="false"
              role="img"
            >
              <title
                id="lds-icon-InfoFill-title-123456789"
              >
                info
              </title>
              <desc
                id="lds-icon-InfoFill-description-123456789"
              />
              <svg
                viewBox="0 0 256 256"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M128,24A104,104,0,1,0,232,128,104.11,104.11,0,0,0,128,24Zm-4,48a12,12,0,1,1-12,12A12,12,0,0,1,124,72Zm12,112a16,16,0,0,1-16-16V128a8,8,0,0,1,0-16,16,16,0,0,1,16,16v40a8,8,0,0,1,0,16Z"
                />
              </svg>
            </svg>
          </div>
          <span
            class="alertContent"
          >
            Your next appointment is today.
             
            <a
              aria-label="View your appointments"
              class="lds-link upcomingAppointmentAlertLink"
              data-testid="view-appointments-banner-link"
            >
              View your appointments
            </a>
          </span>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`UpcomingAppointmentAlert Component should show upcoming appointment alert banner for 1 day from now 1`] = `
<div>
  <div
    class="primary"
    id="test"
  >
    <div
      aria-live="polite"
      class="lds-alert info"
      id="upcoming-appointment-alert"
      role="alert"
    >
      <div
        class="lds-alert-message"
      >
        <button
          class="lds-button-base lds-alert-button"
          data-testid="dismiss-upcoming-appointment-banner"
          type="button"
        >
          <svg
            aria-labelledby="lds-icon-X-title-123456789 lds-icon-X-description-123456789"
            class="lds-icon icon-X inline closeIconWhite"
            focusable="false"
            role="img"
          >
            <title
              id="lds-icon-X-title-123456789"
            >
              close
            </title>
            <desc
              id="lds-icon-X-description-123456789"
            />
            <svg
              viewBox="0 0 256 256"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M205.66,194.34a8,8,0,0,1-11.32,11.32L128,139.31,61.66,205.66a8,8,0,0,1-11.32-11.32L116.69,128,50.34,61.66A8,8,0,0,1,61.66,50.34L128,116.69l66.34-66.35a8,8,0,0,1,11.32,11.32L139.31,128Z"
              />
            </svg>
          </svg>
        </button>
        <div
          class="alert"
        >
          <div
            class="infoIcon"
          >
            <svg
              aria-labelledby="lds-icon-InfoFill-title-123456789 lds-icon-InfoFill-description-123456789"
              class="lds-icon icon-InfoFill"
              classes="lds-alert-icon info"
              focusable="false"
              role="img"
            >
              <title
                id="lds-icon-InfoFill-title-123456789"
              >
                info
              </title>
              <desc
                id="lds-icon-InfoFill-description-123456789"
              />
              <svg
                viewBox="0 0 256 256"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M128,24A104,104,0,1,0,232,128,104.11,104.11,0,0,0,128,24Zm-4,48a12,12,0,1,1-12,12A12,12,0,0,1,124,72Zm12,112a16,16,0,0,1-16-16V128a8,8,0,0,1,0-16,16,16,0,0,1,16,16v40a8,8,0,0,1,0,16Z"
                />
              </svg>
            </svg>
          </div>
          <span
            class="alertContent"
          >
            Your next appointment is in 1 day.
             
            <a
              aria-label="View your appointments"
              class="lds-link upcomingAppointmentAlertLink"
              data-testid="view-appointments-banner-link"
            >
              View your appointments
            </a>
          </span>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`UpcomingAppointmentAlert Component should show upcoming appointment alert banner for 14 days from now 1`] = `
<div>
  <div
    class="primary"
    id="test"
  >
    <div
      aria-live="polite"
      class="lds-alert info"
      id="upcoming-appointment-alert"
      role="alert"
    >
      <div
        class="lds-alert-message"
      >
        <button
          class="lds-button-base lds-alert-button"
          data-testid="dismiss-upcoming-appointment-banner"
          type="button"
        >
          <svg
            aria-labelledby="lds-icon-X-title-123456789 lds-icon-X-description-123456789"
            class="lds-icon icon-X inline closeIconWhite"
            focusable="false"
            role="img"
          >
            <title
              id="lds-icon-X-title-123456789"
            >
              close
            </title>
            <desc
              id="lds-icon-X-description-123456789"
            />
            <svg
              viewBox="0 0 256 256"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M205.66,194.34a8,8,0,0,1-11.32,11.32L128,139.31,61.66,205.66a8,8,0,0,1-11.32-11.32L116.69,128,50.34,61.66A8,8,0,0,1,61.66,50.34L128,116.69l66.34-66.35a8,8,0,0,1,11.32,11.32L139.31,128Z"
              />
            </svg>
          </svg>
        </button>
        <div
          class="alert"
        >
          <div
            class="infoIcon"
          >
            <svg
              aria-labelledby="lds-icon-InfoFill-title-123456789 lds-icon-InfoFill-description-123456789"
              class="lds-icon icon-InfoFill"
              classes="lds-alert-icon info"
              focusable="false"
              role="img"
            >
              <title
                id="lds-icon-InfoFill-title-123456789"
              >
                info
              </title>
              <desc
                id="lds-icon-InfoFill-description-123456789"
              />
              <svg
                viewBox="0 0 256 256"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M128,24A104,104,0,1,0,232,128,104.11,104.11,0,0,0,128,24Zm-4,48a12,12,0,1,1-12,12A12,12,0,0,1,124,72Zm12,112a16,16,0,0,1-16-16V128a8,8,0,0,1,0-16,16,16,0,0,1,16,16v40a8,8,0,0,1,0,16Z"
                />
              </svg>
            </svg>
          </div>
          <span
            class="alertContent"
          >
            Your next appointment is in 14 days.
             
            <a
              aria-label="View your appointments"
              class="lds-link upcomingAppointmentAlertLink"
              data-testid="view-appointments-banner-link"
            >
              View your appointments
            </a>
          </span>
        </div>
      </div>
    </div>
  </div>
</div>
`;
